### PR TITLE
Add schematic for generating UI Extension Point

### PR DESCRIPTION
**Pull request by nvladimirovi**
Implement schematic for extension point generation.
The feature supports the following:
- Generate ext point of type entity action, i.e vm/vapp-action type
- Generate ext point of type primary nav
- Generate ext point of type view only, i.e navigation:application,
navigation:datacenter:* etc.

The schmatic is capable to:
- Gerate basic sources
- Import components or modules in your main module
- Register the extensions in the manifest

Testing Done:
- Generate all kinds of extension points
- Build -> Deploy -> Verifiied!

Signed-off-by: Nikola Vladimirov Iliev <nvladimirovi@vmware.com>

### CODE REVIEW

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    return (host: Tree) => {
+        const workspace = getWorkspace(host);
+        const project = workspace.projects[options.project as string];
+        options.type = options.type ? options.type : "navigation:primary";
+
+        if (!options.path && project) {
+            options.path = buildDefaultPath(project);
+        }
+        
+        const temp = isEntityAction(options) ? "entity-action-extension-point" :
```
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts:31]: What this PR do is to group all types of extension points in 3 main categories - entity-action, primary and view only.
In that way the result categorisation will end up like:
- entity-action template:
     - vm-action
     - vapp-action
- primary template:
    - navigation:primary
- view only template or component based
    - navigation:datacenter:overview
    - navigation:applications
    - navigation:datacenter:compute
    - navigation:datacenter:network
    - navigation:datacenter:storage

When you run the schematic and let's say you want **navigation:applications** extension point the **view only** template will be used to generate that, the same is applicable for the rest extension points.
TLDR: There are 3 base templates which are reused among 9 extension point types.

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        if (!options.manifest) {
+            return host;
+        }
+
+        return updateJsonFile(host, options.manifest, (manifest: Manifest) => {
+            if (!manifest.extensionPoints || !manifest.extensionPoints.length) {
+                manifest.extensionPoints = [];
+            }
+
+            if (manifest.containerVersion !== "9.1.0") {
```
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts:147]: Currently the container version is not leveraged in anyway neither by vcd nor the sdk, however in VCD "9.5.0" container version is used like flag to enable the extension points.

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/schema.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        "project": {
+            "type": "string",
+            "minLength": 1,
+            "default": "cloud-director-container",
+            "x-prompt": "What is the name of your project?"
+        },
+        "name": {
+            "type": "string",
+            "minLength": 1,
+            "x-prompt": "What is the name of your Extension Point Name?"
```
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/schematics/src/schematics/extension-point/schema.json]: I'll fix the typo.

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts]: The check is wrong it has to be !== "9.**5**.0", I'll fix that.

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvladimirovi) [reply packages/angular/projects/vcd/schematics/src/schematics/extension-point/index.ts]: I'll refactor the code as we discussed offline, with config files per ext point.

File: packages/angular/projects/vcd/schematics/src/schematics/extension-point/configs/navigation-datacenter-and-application.config.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import {
+    createExtensionPointFiles,
+    importExtensionPointComponentToModule,
+    updateUiPluginManifest,
+    addExports,
+} from "../schematic-steps";
+import { ExtensionPointDefinision } from "../interfaces";
+
+export const NAVIGATION_DATACENTER_AND_APPLICATION_CONFIG: ExtensionPointDefinision = {
```
Comment (mslavov) [inline packages/angular/projects/vcd/schematics/src/schematics/extension-point/configs/navigation-datacenter-and-application.config.ts:9]: Typo: `definision` -> `definition`

**Review by mslavov — APPROVED** [review]
